shell/src/test/java/tachyon/shell/TFsShellTest.java

```diff
@@ -658,4 +647,238 @@ public void duTest() throws IOException {
+    Assert.assertNull(mTfs.getFile(new TachyonURI("/testWildCards/bar")));
+    Assert.assertNull(mTfs.getFile(new TachyonURI("/testWildCards/bar/foobar3")));
+    Assert.assertNotNull(mTfs.getFile(new TachyonURI("/testWildCards/foobar4")));
+    
+    mFsShell.run(new String[] {"rmr", "/testWildCards/*"});
+    Assert.assertNull(mTfs.getFile(new TachyonURI("/testWildCards/bar")));
+    Assert.assertNull(mTfs.getFile(new TachyonURI("/testWildCards/foo")));
+    Assert.assertNull(mTfs.getFile(new TachyonURI("/testWildCards/foobar4")));    
+  }
+  
+  @Test
+  public void tailWildcardTest() throws IOException {
+    TFsShellUtilsTest.resetTachyonFileHierarchy(mTfs, WriteType.MUST_CACHE);
+    
+    // the expect contents
+    byte[] exp1 = BufferUtils.getIncreasingByteArray(10);
+    byte[] exp2 = BufferUtils.getIncreasingByteArray(20);
+    byte[] exp3 = BufferUtils.getIncreasingByteArray(30);
+    byte[] expect = new byte[exp1.length + exp2.length + exp3.length];
+    System.arraycopy(exp1, 0, expect, 0,                         exp1.length);
+    System.arraycopy(exp2, 0, expect, exp1.length,               exp2.length);
+    System.arraycopy(exp3, 0, expect, exp1.length + exp2.length, exp3.length);
+    
+    int ret = mFsShell.run(new String[] {"tail", "/testWildCards/*/foo*"});
+    Assert.assertEquals(ret, 0);
+    Assert.assertArrayEquals(mOutput.toByteArray(), expect);
+  }
+  
+  @Test
+  public void fileinfoWildCardTest() throws IOException {
+    TFsShellUtilsTest.resetTachyonFileHierarchy(mTfs, WriteType.MUST_CACHE);
+    
+    mFsShell.run(new String[] {"fileinfo", "/testWildCards/*"});
+    String res1 = mOutput.toString();
+    Assert.assertEquals(res1.contains("/testWildCards/foo"), true);
+    Assert.assertEquals(res1.contains("/testWildCards/bar"), true);
+    Assert.assertEquals(res1.contains("/testWildCards/foobar4"), true);
+    Assert.assertEquals(res1.contains("/testWildCards/foo/foobar1"), false);
+    Assert.assertEquals(res1.contains("/testWildCards/bar/foobar3"), false);
+
+    mFsShell.run(new String[] {"fileinfo", "/testWildCards/*/foo*"});
+    String res2 = mOutput.toString();
+    res2 = res2.replace(res1, "");
+    Assert.assertEquals(res2.contains("/testWildCards/foo/foobar1"), true);
+    Assert.assertEquals(res2.contains("/testWildCards/foo/foobar2"), true);
+    Assert.assertEquals(res2.contains("/testWildCards/bar/foobar3"), true);
+    Assert.assertEquals(res2.contains("/testWildCards/foobar4"), false);
+  }
+  
+  @Test
+  public void reportWildcardTest() throws IOException {
+    TFsShellUtilsTest.resetTachyonFileHierarchy(mTfs, WriteType.MUST_CACHE);
+    
+    mFsShell.run(new String[] {"report", "/testWildCards/*/foo*"});
+    String expectedOutput = "";
+    expectedOutput += "/testWildCards/foo/foobar1 with file id " 
+        + mTfs.getFileId(new TachyonURI("/testWildCards/foo/foobar1")) 
+        + " has reported been report lost.\n";
+    expectedOutput += "/testWildCards/foo/foobar2 with file id " 
```

Comment:
Or we can have a `String[]` containing the paths like `/testWildCards/foo/foobar1, /testWildCards/foo/foobar2....`, then iterate over the array and append output to `expectedOutput`, then we can get rid of the redundant codes.